frontend/src/Components/Chatbox.tsx

```diff
@@ -254,17 +249,17 @@ export default function Chatbox({
     });
 
     return () => {
-      if (socket) {
-        socket.off("message");
-        socket.off("delete_message");
-        socket.off("report_applied");
-        socket.off("ban_applied");
-        socket.off("unban_applied");
-        socket.off("join_room_resp");
-        socket.off("report_user_resp");
+      if (socketTracker) {
+        socketTracker.unsubscribe("message");
+        socketTracker.unsubscribe("delete_message");
+        socketTracker.unsubscribe("report_applied");
+        socketTracker.unsubscribe("ban_applied");
+        socketTracker.unsubscribe("unban_applied");
+        socketTracker.unsubscribe("join_room_resp");
+        socketTracker.unsubscribe("report_user_resp");
       }
     };
-  }, [socket, userData.username]);
+  }, [socketTracker, userData.username]);
```

Comment:
Let's only resubscribe when socketTracker changes! 

```suggestion
  }, [socketTracker]);
```